src/map/utils/battleutils.cpp

```diff
@@ -5820,10 +5821,18 @@ namespace battleutils
                 }
                 else
                 {
-                    // draw in!
-                    PMember->loc.p.x = nearEntity.x;
-                    PMember->loc.p.y = nearEntity.y;
-                    PMember->loc.p.z = nearEntity.z;
+                    if (toFrontOfMob)
+                    {
+                        PMember->loc.p.x = nearEntity.x;
```

Comment:
Would it make sense to calculate these positions at the very start of the draw in, validate them on the navmesh, and then assign them here?

Reply:
> Would it make sense to calculate these positions at the very start of the draw in, validate them on the navmesh, and then assign them here?

I think the function is already doing what you describe? It calculates and validates near the start of the draw-in function. Or do you mean even earlier in the draw-in process like in a different function?